internal/stolonctl/pkg/kubernetes/status.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright (C) 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"fmt"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/util/node"
+)
+
+// PodStatus represents status of a kubernetes Pod
+type PodStatus struct {
+	// Pod name
+	Name string
+	// IP address of the host to which the pod is assigned
+	HostIP string
+	// IP address allocated to the pod
+	PodIP string
+	// Status of the pod
+	Status string
+	// ReadyContainers - number of ready containers
+	ReadyContainers int
+	// TotalContainers - total number of containers in Pod
+	TotalContainers int
+	// CreationTimestamp is a timestamp representing the server time
+	// when this pod was created
+	CreationTimestamp *metav1.Time
+}
+
+// DeterminePodStatus returns state of the pod, value of total containers
```

Comment:
which would not be necessary had you named the returns:
```go
func DeterminePodStatus(pod v1.Pod) (podStatus string, totalContainers int, readyContainers int) {
```
This provides the necessary details for the caller.
I would also use `number` instead of `value` as it reads better.